Support parallel mock device mappers

simonjbeaumont opened the pull request.

simonjbeaumont committed:
Fix compilier warnings

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>

simonjbeaumont committed:
Enable error on warning

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>

simonjbeaumont committed:
CP-14357: mock: Remove dead code (unused and hidden by interface)

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>

simonjbeaumont committed:
CP-14357: mock: Replace in-memory (with write-through) with RMW

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>

simonjbeaumont committed:
CP-14357: Factor out read modify write into helper function

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>

simonjbeaumont committed:
oasis: Separate BuildDepends for clarity

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>

simonjbeaumont committed:
CP-14357: mock: Add locking around read-modify-write

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>

simonjbeaumont committed:
CP-14357: test: Check thread safety of mock DM

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>

simonjbeaumont committed:
CP-14357: test: Correct filename canonicalisation

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>

simonjbeaumont committed:
CP-14357: test: Check process safety of mock DM

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>

simonjbeaumont committed:
CP-14357: mock: Allow parallel execution across processes

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>

simonjbeaumont committed:
test: Remove mock serialisation test since we're always doing this

Now the mock devmapper uses a read modify write loop, every use of it is
testing that we can serialise and deserialise the devices.

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>

simonjbeaumont committed:
CP-14357: mock: Support multiple file backends

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>

jonludlam commented:
OK, looks good to me, so long as Travis is happy :-)

We may need to update our RPM BuildRequires too.

simonjbeaumont committed:
opam: Require cmdliner for parallel test runner

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>

simonjbeaumont committed:
Restructure project

Now it is compatible with opam and standard packaging. We have still avoided
commiting the oasis autogen files. But now we have gone with:

* A straight-forward ./configure script;
* Having the tests we want in the _oasis file;
* Having tests not unilaterally enabled;
* Using an autogeneated (simpler) Makefile;
* An opam file that includes fixes for the errors from `opam lint` (below).

OPAM lint errors:

    $ opam lint
    Errors found in /local/work/code/camldm/opam:
                error 25: Missing field 'authors'
                error 33: Field 'os' is deprecated, use 'available' and the 'os'
                  variable instead
                error 35: Missing field 'homepage'
              warning 36: Missing field 'bug-reports'
              warning 37: Missing field 'dev-repo'

Authors filled in by using what `git` provides:

    $ git shortlog -sn
        72  David Scott
        20  Si Beaumont
         9  Dave Scott
         9  Jon Ludlam
         3  John Else
         2  Kaifeng Zhu

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>

simonjbeaumont commented:
@jonludlam... looks like I've finally managed to placate Travis. I think the project is a bit more sane for it too.

simonjbeaumont committed:
Release 0.11.0

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>
